src/overlays/actors/ovl_En_Fire_Rock/z_en_fire_rock.c

```diff
@@ -55,27 +61,337 @@ static ColliderCylinderInit D_80A12CA0 = {
     },
     {
         ELEMTYPE_UNK0,
-        { 0xFFCFFFFF, 0x09, 0x08 },
+        { 0xFFCFFFFF, 0x01, 0x08 },
         { 0xFFCFFFFF, 0x00, 0x00 },
         TOUCH_ON | TOUCH_SFX_NORMAL,
         BUMP_ON,
         OCELEM_NONE,
     },
     { 30, 30, -10, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fire_Rock/EnFireRock_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Fire_Rock/EnFireRock_Destroy.s")
+void EnFireRock_Init(Actor* thisx, GlobalContext* globalCtx) {
+
+    GlobalContext* globalCtx2 = globalCtx;
+    Player* player = PLAYER;
+    EnFireRock* this = THIS;
+    s16 temp;
+
+    this->type = this->actor.params;
+    if (this->type != FIRE_ROCK_CEILING_SPOT_SPAWNER) {
+        ActorShape_Init(&thisx->shape, 0.0f, ActorShadow_DrawCircle, 15.0f);
+        if (this->type != FIRE_ROCK_ON_FLOOR) {
+            this->angularVelocity.x = (Rand_ZeroFloat(10.0f) + 15.0f);
+            this->angularVelocity.y = (Rand_ZeroFloat(10.0f) + 15.0f);
+            this->angularVelocity.z = (Rand_ZeroFloat(10.0f) + 15.0f);
+        }
+    }
+    switch (this->type) {
+        case FIRE_ROCK_CEILING_SPOT_SPAWNER:
+            this->actor.draw = NULL;
+            // "☆☆☆☆☆ ceiling waiting rock ☆☆☆☆☆"
+            osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 天井待ち岩 ☆☆☆☆☆ \n" VT_RST);
+            this->actionFunc = FireRock_WaitSpawnRocksFromCeiling;
+            break;
+        case FIRE_ROCK_ON_FLOOR:
+            Actor_SetScale(&this->actor, 0.03f);
+            Collider_InitCylinder(globalCtx, &this->collider);
+            Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &D_80A12CCC);
+            // "☆☆☆☆☆ floor rock ☆☆☆☆☆"
+            osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 床岩 ☆☆☆☆☆ \n" VT_RST);
+            this->collider.dim.radius = 23;
+            this->collider.dim.height = 37;
+            this->collider.dim.yShift = -10;
+            Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, (u8)6U);
```

Comment:
```suggestion
            Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, 6);
```
Unless its needed of course.

Reply:
this should use the actor category enum